client/src/lib/content/hooks/useGetExercisesByTags.tsx

```diff
@@ -0,0 +1,28 @@
+import {intersection, take} from 'ramda';
+import {useMemo} from 'react';
+import useAppState from '../../appState/state/state';
+import useExercises from './useExercises';
+import {Tag} from '../../../../../shared/src/types/generated/Tag';
+
+const useGetExercisesByTags = (tags: Tag[], excludeId?: string) => {
+  const {showHiddenContent} = useAppState(state => state.settings);
+  const exercises = useExercises();
+  const tagNames = useMemo(() => tags?.map(t => t.tag), [tags]);
+
+  return useMemo(() => {
+    if (!tagNames) return [];
+    const exerciseList = exercises
+      .filter(e => showHiddenContent || !e.hidden)
```

Comment:
This filter is unnecessary because it's handled in i18n. If you see hidden content even though the toggle is off, try restarting the app.